packages/pushnotification/android/src/main/java/com/amazonaws/amplify/pushnotification/RNPushNotificationModule.java

```diff
@@ -64,9 +67,22 @@ public void initialize() {
     }
 
     @ReactMethod
-    public void getToken(Callback callback) {
-        String token =  FirebaseInstanceId.getInstance().getToken();
-        Log.i(LOG_TAG, "getting token" + token);
-        callback.invoke(token);
+    public void getToken(final Callback callback) {
+        final Task<String> taskToken =  FirebaseMessaging.getInstance().getToken();
+            taskToken.addOnCompleteListener(new OnCompleteListener<String>() {
+                @Override
+                public void onComplete(@NonNull Task<String> task) {
+                    if (task.isSuccessful()) {
+                        String token = task.getResult();
+                        Log.i(LOG_TAG, "getting token " + token);
```

Comment:
Nit: If successful, maybe "got token" is more accurate here?
```suggestion
                        Log.i(LOG_TAG, "got token " + token);
```